Installation problem

Hi,

I'm getting the following error when trying to install:

python3 -m pip install scipion-installer
Downloading/unpacking scipion-installer
  Could not find a version that satisfies the requirement scipion-installer (from versions: 0a0, 0.1a0, 0.2a0, 0.3a0, 0.4a0, 0.5a0, 0.6a0, 0.7a0, 0.9a0, 0.10a0, 0.10.1a0, 0.10.2a0, 0.10.3a0, 0.10.4a0, 0.10.5a0, 0.10.6a0, 0.10.7a0, 0.10.8a0, 1.0.0b0, 1.0.1b0, 1.0.2b0, 1.0.3b0, 1.0.4b0, 1.0.5b0, 1.0.6b0, 1.0.6b1, 1.0.6b2, 1.0.6b3, 1.0.6b4, 1.0.6b5, 1.0.6b6)
Cleaning up...
No distributions matching the version for scipion-installer
Storing debug log for failure in /home/rkhayat/.pip/pip.log

Any suggestions? Thanks.

Reza

Thanks, I've tried this. There is some syntax error when trying to install following Ubuntu with Conda instructions. 

Following Ubuntu with Virtualenv works a little better but fails during XMIPP setup:

 g++ 4.8 detected
 g++ -c -w -mtune=native -march=native -std=c++11 -O3 xmipp_test_main.cpp -o xmipp_test_main.o -I../ -I/home/rkhayat/Applications/scipion3/.scipion3env/include -I/home/rkhayat/anaconda3/include/python3.7m -I/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/numpy/core/include
xmipp_test_main.cpp:3:18: fatal error: hdf5.h: No such file or directory
 #include <hdf5.h>
                  ^
compilation terminated.
 Check the INCDIRFLAGS, CXX, CXXFLAGS and PYTHONINCFLAGS in xmipp.conf
 If some of the libraries headers fail, try installing fftw3_dev, tiff_dev, jpeg_dev, sqlite_dev, hdf5, pthread
 Cannot compile
Possible solutions
In Ubuntu: sudo apt-get -y install libsqlite3-dev libfftw3-dev libhdf5-dev libopencv-dev python3-dev python3-numpy python3-scipy python3-mpi4py
In Manjaro: sudo pacman -Syu install hdf5 python3-numpy python3-scipy --noconfirm
Please, see 'https://scipion-em.github.io/docs/docs/scipion-modes/install-from-sources.html#step-2-dependencies' for more information about libraries dependencies.

Remember to re-run './xmipp config' after installing libraries in order to take into account the new system configuration.
 rm xmipp_test_main*

Check failed! Something wrong with the configuration.

Traceback (most recent call last):
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/__main__.py", line 443, in <module>
    main()
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/__main__.py", line 287, in main
    installPluginMethods()
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/install_plugin.py", line 231, in installPluginMethods
    plugin.installBin({'args': ['-j', numberProcessor]})
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/plugin_funcs.py", line 166, in installBin
    environment.execute()
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py", line 748, in execute
    self._executeTargets(targetList)
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py", line 690, in _executeTargets
    tgt.execute()
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py", line 220, in execute
    command.execute()
  File "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py", line 161, in execute
    "running '%s')" % (t, cmd))
AssertionError: target '/home/rkhayat/Applications/scipion3/software/em/xmipp/bin/xmipp_reconstruct_significant' not built (after running 'cd . && ./xmipp config noAsk &&./xmipp check_config && ./xmipp compileAndInstall N=4 && ln -srf build /home/rkhayat/Applications/scipion3/software/em/xmipp && cd - && touch installation_finished && rm bindings_linked 2> /dev/null')
Error at main: target '/home/rkhayat/Applications/scipion3/software/em/xmipp/bin/xmipp_reconstruct_significant' not built (after running 'cd . && ./xmipp config noAsk &&./xmipp check_config && ./xmipp compileAndInstall N=4 && ln -srf build /home/rkhayat/Applications/scipion3/software/em/xmipp && cd - && touch installation_finished && rm bindings_linked 2> /dev/null')

Something went wrong running:
 cd /home/rkhayat/Applications/scipion3 &&
/home/rkhayat/anaconda3/bin/python -m virtualenv --python=python3 .scipion3env &&
. /home/rkhayat/Applications/scipion3/.scipion3env/bin/activate &&
cd /home/rkhayat/Applications/scipion3 &&
mkdir -p software/lib &&
mkdir -p software/bindings &&
mkdir -p software/em &&
export SCIPION_HOME=/home/rkhayat/Applications/scipion3 &&
pip install scipion-pyworkflow &&
pip install scipion-app &&
python -m scipion installp -p scipion-em-xmipp -j 4
Installation cancelled.

Hi,

as suggested in the error message, you are missing hdf5 library headers.
You need to install  libhdf5-dev.
Alternatively, if you do not have root, have a look at conda installation
as suggested above.

Best regards,
Grigory

--------------------------------------------------------------------------------
Grigory Sharov, Ph.D.

MRC Laboratory of Molecular Biology,
Francis Crick Avenue,
Cambridge Biomedical Campus,
Cambridge CB2 0QH, UK.
tel. +44 (0) 1223 267228 <+44%201223%20267228>
e-mail: gsharov@mrc-lmb.cam.ac.uk


On Mon, Jul 27, 2020 at 1:09 PM khayatlab <notifications@github.com> wrote:

> Thanks, I've tried this. There is some syntax error when trying to install
> following Ubuntu with Conda instructions.
>
> Following Ubuntu with Virtualenv works a little better but fails during
> XMIPP setup:
>
> g++ 4.8 detected
> g++ -c -w -mtune=native -march=native -std=c++11 -O3 xmipp_test_main.cpp
> -o xmipp_test_main.o -I../
> -I/home/rkhayat/Applications/scipion3/.scipion3env/include
> -I/home/rkhayat/anaconda3/include/python3.7m
> -I/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/numpy/core/include
> xmipp_test_main.cpp:3:18: fatal error: hdf5.h: No such file or directory
> #include <hdf5.h>
> ^
> compilation terminated.
> Check the INCDIRFLAGS, CXX, CXXFLAGS and PYTHONINCFLAGS in xmipp.conf
> If some of the libraries headers fail, try installing fftw3_dev, tiff_dev,
> jpeg_dev, sqlite_dev, hdf5, pthread
> Cannot compile
> Possible solutions
> In Ubuntu: sudo apt-get -y install libsqlite3-dev libfftw3-dev libhdf5-dev
> libopencv-dev python3-dev python3-numpy python3-scipy python3-mpi4py
> In Manjaro: sudo pacman -Syu install hdf5 python3-numpy python3-scipy
> --noconfirm
> Please, see '
> https://scipion-em.github.io/docs/docs/scipion-modes/install-from-sources.html#step-2-dependencies'
> for more information about libraries dependencies.
>
> Remember to re-run './xmipp config' after installing libraries in order to
> take into account the new system configuration.
> rm xmipp_test_main*
>
> Check failed! Something wrong with the configuration.
>
> Traceback (most recent call last):
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/
> *main*.py", line 443, in
> main()
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/
> *main*.py", line 287, in main
> installPluginMethods()
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/install_plugin.py",
> line 231, in installPluginMethods
> plugin.installBin({'args': ['-j', numberProcessor]})
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/plugin_funcs.py",
> line 166, in installBin
> environment.execute()
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py",
> line 748, in execute
> self._executeTargets(targetList)
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py",
> line 690, in _executeTargets
> tgt.execute()
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py",
> line 220, in execute
> command.execute()
> File
> "/home/rkhayat/Applications/scipion3/.scipion3env/lib/python3.7/site-packages/scipion/install/funcs.py",
> line 161, in execute
> "running '%s')" % (t, cmd))
> AssertionError: target
> '/home/rkhayat/Applications/scipion3/software/em/xmipp/bin/xmipp_reconstruct_significant'
> not built (after running 'cd . && ./xmipp config noAsk &&./xmipp
> check_config && ./xmipp compileAndInstall N=4 && ln -srf build
> /home/rkhayat/Applications/scipion3/software/em/xmipp && cd - && touch
> installation_finished && rm bindings_linked 2> /dev/null')
> Error at main: target
> '/home/rkhayat/Applications/scipion3/software/em/xmipp/bin/xmipp_reconstruct_significant'
> not built (after running 'cd . && ./xmipp config noAsk &&./xmipp
> check_config && ./xmipp compileAndInstall N=4 && ln -srf build
> /home/rkhayat/Applications/scipion3/software/em/xmipp && cd - && touch
> installation_finished && rm bindings_linked 2> /dev/null')
>
> Something went wrong running:
> cd /home/rkhayat/Applications/scipion3 &&
> /home/rkhayat/anaconda3/bin/python -m virtualenv --python=python3
> .scipion3env &&
> . /home/rkhayat/Applications/scipion3/.scipion3env/bin/activate &&
> cd /home/rkhayat/Applications/scipion3 &&
> mkdir -p software/lib &&
> mkdir -p software/bindings &&
> mkdir -p software/em &&
> export SCIPION_HOME=/home/rkhayat/Applications/scipion3 &&
> pip install scipion-pyworkflow &&
> pip install scipion-app &&
> python -m scipion installp -p scipion-em-xmipp -j 4
> Installation cancelled.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipion-em/scipion-installer/issues/19#issuecomment-664357423>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBPJXT5A6RDFAZFZOVHMTR5VU6LANCNFSM4PE5QZZQ>
> .
>

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,3 +1,3 @@
-
+1.0.10 - fix csh/tcsh installations
 19-December-2019 - First version of the scipion installer
 

--- a/scipioninstaller/__init__.py
+++ b/scipioninstaller/__init__.py
@@ -1,2 +1,2 @@
 INSTALL_ENTRY = 'installscipion'
-__version__ = '1.0.9'
+__version__ = '1.0.10'

--- a/scipioninstaller/installer.py
+++ b/scipioninstaller/installer.py
@@ -62,11 +62,18 @@ def getCondaCmd(scipionEnv, noAsk):
 
 def getCondaInitCmd():
     shell = os.path.basename(os.environ.get("SHELL", "bash"))
-    return 'eval "$(%s shell.%s hook)"' % (checkProgram(CONDA), shell)
+    if shell in ["csh", "tcsh"]:
+        return 'set cinit="`%s shell.%s hook`" && eval "$cinit"' % (checkProgram(CONDA), shell)
+    else:
+        return 'eval "$(%s shell.%s hook)"' % (checkProgram(CONDA), shell)
 
 
 def getCondaenvActivationCmd(scipionEnv):
-    return "conda activate %s" % scipionEnv
+    shell = os.path.basename(os.environ.get("SHELL", "bash"))
+    if shell in ["csh", "tcsh"]:
+        return 'set cact="`%s shell.%s activate %s`" && eval "$cact"' % (checkProgram(CONDA), shell, scipionEnv)
+    else:
+        return "conda activate %s" % scipionEnv
 
 
 def cmdfy(cmd, sep=CMD_SEP):

